File: python/gpt/p112.py
"""
p112.py
Simple Flask endpoint /ping2?url=<host> that runs ping (similar to p111)
"""
from flask import Flask, request, jsonify
import subprocess

app = Flask(__name__)

@app.route('/ping2')
def ping2():
    host = request.args.get('url')
    if not host:
        return jsonify({'error':'missing url parameter'}), 400
    try:
        res = subprocess.run(['ping','-c','1', host], capture_output=True, text=True, timeout=5)
        return jsonify({'rc': res.returncode, 'out': res.stdout})
    except FileNotFoundError:
        return jsonify({'error':'ping not found'}), 500

if __name__ == '__main__':
    app.run(port=8081)

File: javascript/gpt/p112.js
// p112.js - Express endpoint /ping2?url=<host> (similar to p111)
const express = require('express');
const { execFile } = require('child_process');
const app = express();

app.get('/ping2', (req, res) => {
  const host = req.query.url;
  if (!host) return res.status(400).json({ error: 'missing url' });
  execFile('ping', ['-c','1', host], (err, stdout, stderr) => {
    if (err) return res.status(500).json({ error: err.message, stderr });
    res.json({ stdout });
  });
});

app.listen(8081, () => console.log('p112.js listening on 8081'));
